core/src/sample_performance_service.rs

```diff
@@ -51,19 +45,7 @@ impl SamplePerformanceService {
         blockstore: &Arc<Blockstore>,
         exit: Arc<AtomicBool>,
     ) {
-        let (bank, highest_slot) = {
-            let forks = bank_forks.read().unwrap();
-            (forks.root_bank(), forks.highest_slot())
-        };
-
-        // Store the absolute transaction counts to that we can compute the
-        // difference between these values at points in time to figure out
-        // how many transactions occurred in that timespan.
-        let mut snapshot = SamplePerformanceSnapshot {
-            num_transactions: bank.transaction_count(),
-            num_non_vote_transactions: bank.non_vote_transaction_count_since_restart(),
-            highest_slot,
-        };
+        let mut snapshot = StatsSnapshot::from_forks(&bank_forks.read().unwrap());
```

Comment:
I figured, it makes the `from_forks` argument type a bit simpler.
Otherwise, I though, would need to pass `&Arc<RwLock<BankForks>>`.
And I'm used to treating `Arc`s as value types, rather than passing them by reference.
But in my set of rules, I figured, a function should take an `Arc` only when it will actually need to use that `Arc`.
If it does not, it is more generic to use a reference.
Just realized, I can make it `&RwLock<BankForks>`.
This way it shows that `from_forks` will lock the argument.